crates/red_knot_python_semantic/resources/mdtest/metaclass.md

```diff
@@ -0,0 +1,139 @@
+## Default
+
+```py
+class C: ...
+
+reveal_type(C.__class__)  # revealed: type
+```
+
+## Basic
+
+```py
+class A: ...
+
+class B(metaclass=A): ...
+
+reveal_type(B.__class__)  # revealed: Literal[A]
```

Comment:
Should this be `A` rather than `Literal[A]`?

Reply:
(I don't feel confident about my understanding and usage of `ClassLiteralType` vs. `Class`.)

Reply:
This should be `Literal[A]`!

You're confusing three things here, I think. Remember that every type is a set of possible runtime values.
- All `ClassLiteralType`s are sets of size `. The sole member of the set described by `Literal[A]` (which is a class-literal type) is the class object `A` itself.
- `InstanceType`s are sets of (usually) unknown size. The set described by the type `A` (which is an instance type) is the set of "all possible instances of `A` that could ever exist". Note that this is a disjoint set from the set described by `Literal[A]`: the class object `A` is not an instance of `A`.
- We have a struct named `Class` that is used as the inner data for both `InstanceType` and `ClassLiteralType`. But this struct doesn't represent a type at all; it's just a "bag of data" that you can query about a specific class definition at runtime.